### PR TITLE
[doc] Refresh components.md

### PR DESCRIPTION
**Pull request by ro0gr**

### CODE REVIEW

File: guides/components.md [hunk 1]
```diff
@@ -134,96 +98,145 @@ the following configuration will match the article paragraph element
...
-  textBody: text('p'),
+  textBody: {
+    scope: 'p'
+  }
 });
 
-andThen(function() {
-  assert.equal(page.textBody, 'Lorem ipsum dolor.');
-});
+assert.equal(page.textBody.text, 'Lorem ipsum dolor.');
```
[the remark this answers is not in the document]
Comment (ro0gr) [reply guides/components.md]: > in the chain of page objects

I think using a "chain" here can be confusing. What about "in the **hierarchy** of page objects"?

**Comment by ro0gr** [comment]
@mistahenry you rule 🚀💯

**Comment by ro0gr** [comment]
Going to review it tomorrow with a fresh mind and decide whether the calculator example would be applicable here(probably in a form of the final boss)..

File: guides/components.md [hunk 2]
```diff
@@ -134,96 +98,145 @@ the following configuration will match the article paragraph element
...
-const page = create({
-  input: {
-    scope: '#userName',
+Attributes are just Page Object aware wrappers around the low level DOM operations. It allows you to configure component's behavior in a declarative fashion.
 
-    hasError: hasClass('has-error'),
-    value: value(),
-    fillIn: fillable()
-  },
+By default attribute uses a parent component's `scope`:
```
[the remark this answers is not in the document]
Comment (mistahenry) [reply guides/components.md]: @ro0gr This change is still needed

File: guides/components.md [hunk 3]
```diff
@@ -134,96 +96,113 @@ the following configuration will match the article paragraph element
...
 });
 
-andThen(function() {
-  assert.equal(page.textBody, 'Lorem ipsum dolor.');
-});
+assert.equal(page.textBody.text, 'Lorem ipsum dolor.');
 ```
 
-The attribute's selector can be omited when the scope matches the element we want to use.
+When `page.textBody.text` is evaluated, each link in the hierarchy of page objects has its scope incorporated into the final selector thats used to find an element whose text is returned (`.article p` in this example).
```
Comment (mistahenry) [inline guides/components.md]: When `page.textBody.text` is evaluated, each link in the hierarchy of page objects has its scope incorporated into the final selector that[']s used to find an element whose text is returned (`.article p` in this example).

**Comment by ro0gr** [comment]
I think the page can still be improved. However the original intention to refresh it was satisfied in this PR. We can keep improving the page in the future.

**Pull request merged**